pkg/plugins/manager/manager.go

```diff
@@ -226,8 +191,8 @@ func (m *PluginManager) Renderer() *plugins.Plugin {
 }
 
 func (m *PluginManager) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
-	plugin := m.Plugin(req.PluginContext.PluginID)
-	if plugin == nil {
+	plugin, exists := m.plugins[req.PluginContext.PluginID]
```

Comment:
Earlier when using `m.Plugins(...)` there was usage of `m.pluginsMu` for adding a read lock. Also there was a check `p.IsDecommissioned()`. What's the idea, skipping all of that for now?

Reply:
Still yet to add that back in the relevant places - should have mentioned 🤦  Was just curious if you thought the general shape looked sensible!

Reply:
My bad. It was a draft right. general shape looks sensible. Feels like a good opportunity to introduce context as the first argument, but could be done in a separate PR I guess.

Reply:
Yeah was thinking that... let me see if it gets too big by then 😄